src/components/tl/mlx5/tl_mlx5_wqe.c

```diff
@@ -0,0 +1,301 @@
+                                        uint16_t nrows, int send_flags)
+{
+    uint32_t                  opcode   = MLX5_OPCODE_LOCAL_MMO;
+    uint32_t                  opmode   = 0x0; //TRanspose
+    uint32_t                  n_ds     = 4;
+    struct ibv_qp_ex *        qp_ex    = ibv_qp_to_qp_ex(qp);
+    struct mlx5dv_qp_ex *     mqp      = mlx5dv_qp_ex_from_ibv_qp_ex(qp_ex);
+    int                       fm_ce_se = 0;
+    char                      wqe_desc[n_ds * DS_SIZE];
+    struct mlx5_wqe_ctrl_seg *ctrl;
+    struct mlx5_wqe_data_seg *data;
+    transpose_seg_t *         tseg;
+
+    if (send_flags & IBV_SEND_SIGNALED) {
+        qp_ex->wr_id = 0;
+        fm_ce_se |= MLX5_WQE_CTRL_CQ_UPDATE;
+    }
+
+    memset(wqe_desc, 0, n_ds * DS_SIZE);
+    /* SET CTRL SEG */
+    ctrl = (void *)wqe_desc;
+    mlx5dv_set_ctrl_seg(ctrl, /* pi */ 0x0, opcode, opmode, qp->qp_num,
+                        fm_ce_se, n_ds, 0x0, 0x0);
+
+    /* SET TRANSPOSE SEG */
+    tseg               = PTR_OFFSET(ctrl, DS_SIZE);
+    tseg->element_size = htobe32(element_size);
+    tseg->num_rows     = htobe16(nrows);
+    tseg->num_cols     = htobe16(ncols);
+
+    /* SET SRC DATA SEG */
+    data = PTR_OFFSET(tseg, DS_SIZE);
+    mlx5dv_set_data_seg(data, ncols * nrows * element_size, src_mr_lkey,
+                        src_mkey_addr);
+
+    /* SET DST DATA SEG */
+    data = PTR_OFFSET(data, DS_SIZE);
+    mlx5dv_set_data_seg(data, ncols * nrows * element_size, dst_mr_key,
+                        dst_addr);
+    mlx5dv_wr_raw_wqe(mqp, wqe_desc);
+    return UCC_OK;
+}
+
+/* The strided block format is as the following:
+ * | repeat_block | entry_block | entry_block |...| entry_block |
+ * While the repeat entry contains details on the list of the block_entries. */
+static void umr_pointer_seg_init(uint32_t                      repeat_count,
+                                 uint16_t                      num_interleaved,
+                                 struct mlx5dv_mr_interleaved *data,
+                                 struct mlx5_wqe_umr_pointer_seg *pseg,
+                                 uint32_t ptr_mkey, void *ptr_address,
+                                 int *xlat_size, uint64_t *reglen)
+{
+    uint64_t                              byte_count = 0;
+    struct mlx5_wqe_umr_repeat_block_seg *rb;
+    struct mlx5_wqe_umr_repeat_ent_seg *  eb;
+    int                                   i;
+
+    /* set pointer segment */
+    pseg->mkey    = htobe32(ptr_mkey);
+    pseg->address = htobe64((uint64_t)ptr_address);
+
+    /* set actual repeated and entry blocks segments */
+    rb               = ptr_address;
+    rb->op           = htobe32(0x400); // PRM header entry - repeated blocks
+    rb->reserved     = 0;
+    rb->num_ent      = htobe16(num_interleaved);
+    rb->repeat_count = htobe32(repeat_count);
+    eb               = rb->entries;
+
+    /*
+	 * ------------------------------------------------------------
```

Comment:
comment alignment

Reply:
on my editor the stars are aligned. I think that's just maybe a space character that is rendered as a tab in github. I could align the stars with the slash "/"

Reply:
solved